gpytorch/kernels/distributional_input_kernel.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+
+from torch.distributions import Normal, kl_divergence
+
+from .kernel import Kernel
+
+
+def _symmetrized_kl(dist1, dist2, num_dims=2, eps=1e-8):
+    dist1_mean = dist1[..., :num_dims].unsqueeze(-3)
+    dist1_logvar = dist1[..., num_dims:].unsqueeze(-2)
+    dist1_var = eps + dist1_logvar.exp()
+
+    dist2_mean = dist2[..., :num_dims].unsqueeze(-3)
+    dist2_logvar = dist2[..., num_dims:].unsqueeze(-2)
+    dist2_var = eps + dist2_logvar.exp()
+
+    dist1 = Normal(dist1_mean, dist1_var ** 0.5)
+    dist2 = Normal(dist2_mean, dist2_var ** 0.5)
+
+    res = (kl_divergence(dist1, dist2) + kl_divergence(dist2, dist1)).sum(-1)
```

Comment:
Since we know / assume these are Gaussian, we could just use `_kl_normal_normal` instead to shortcut all the registry business. But I guess not using the "private" API is also virtuous. Alternatively we could just compute this directly without having to construct any distributions:

```
var_ratio12 = dist1_var / dist2_var
log_var_ratio12 = var_ratio12.log()
loc_sqdiffs = (dist1_mean - dist2_mean).pow(2)
kl1 = 0.5 * (var_ratio12 + loc_sqdiffs / dist2_var - 1 - log_var_ratio12)
kl2 = 0.5 * (var_ratio12.reciprocal() + loc_sqdiffs / dist1_var - 1 + log_var_ratio12)
kl = kl1 + kl2
```

Reply:
Switched to your implementation and resolving.